### ISSUE TITLE
Default Docker executables don't work on Apple Silicon with Homebrew

### ISSUE THREAD

**Issue by stefanscheidt** [avast/gradle-docker-compose-plugin#358]
If you use the plugin on Apple hardware with Apple silicon and you use `docker` and `docker-compose` installed via Homebrew (e.g. because you don't want or you can't use the payed Docker Desktop, but e.g. colima), the "sensible defaults" (see https://github.com/avast/gradle-docker-compose-plugin/blob/0.16.4/src/main/groovy/com/avast/gradle/dockercompose/ComposeSettings.groovy#L147-L155) for the Docker executables don't work, because the Homebrew prefix is `/opt/homebrew/bin` on those machines.

Maybe it's worth mentioning the defaults and instructions how to adjust them in the README.

**Comment by augi** [comment]
Hello, do you please know about any way how to detect `docker` and `docker-compose` has been installed via Homebrew? For example, to check an environment variable or presence of a file? I don't know the Apple ecosystem, this is why I'm asking 🙏

**Comment by stefanscheidt** [comment]
Hi. No, I'm sorry. I have no better idea than mentioning the fact in the README. (Probably I'll find the time to submit a PR with a suggestion for the wording if you want to "solve" it like this.) 

Just out of curiosity: What does the code comment "Default installation is inaccessible from path" mean? (I'm not using Docker Desktop for Mac anymore, so I can not check what happens there ...)

**Comment by augi** [comment]
Unfortunately I don't know because I have no experience with MacOS, the author of this part of code is someone else.

**Comment by mfriess2** [comment]
This works as a workaround on MacOS:
```
executable.set("docker-compose") // force the plugin to search in the PATH (instead of "/usr/local/bin")
dockerExecutable.set("docker") // force the plugin to search in the PATH
```

Would be great if that could go into the MacOS specific plugin implementation!
Because I'm not sure if doing it with this workaround will work cross-plattform for Linux and Windows builds.

**Comment by augi** [comment]
Hello, there is code that ensures MacOS-specific default for these variables: https://github.com/avast/gradle-docker-compose-plugin/blob/main/src/main/groovy/com/avast/gradle/dockercompose/ComposeSettings.groovy#L147-L151

As I'm not MacOS user, I'm just not sure what would be better defaults.

**Comment by mfriess2** [comment]
I would propose to remove these MacOS defaults in line 150/151 and use the same defaults across all platforms.
Because:

1) For Rancher Desktop, `docker`/`docker-compose` will be put into the path by default when it starts
2) For Docker Desktop,  `docker`/`docker-compose` will also be in the path (symlinks are put into /usr/local/bin which is itself in the path)
3) Usually you want to have "docker" and/or "docker-compose" in your path anyways to be able to use it on the command line
4) Your plugin documentation currently states: `executable = '/path/to/docker-compose' // allow to set the path of the docker-compose executable (useful if not present in PATH)` which is obviously wrong for MacOS (and got me confused when hitting this behavior)

**Comment by augi** [comment]
Addressed in [0.16.5](https://github.com/avast/gradle-docker-compose-plugin/releases/tag/0.16.5), hope it will help 🤞

**Comment by fvargasj** [comment]
What about applying the same fix as for the version 0.16.5 in the version 0.13 which is supporting Gradle 5? I have the problem that I cannot upgrade the Gradle version and I cannot find any configuration for v0.13.4 of the plugin that avoids the following error message.

`> Task :composeUp FAILED
exec: "docker": executable file not found in $PATH
Current PATH : /usr/bin:/bin:/usr/sbin:/sbin

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':composeUp'.
> Exit-code 1 when calling /usr/local/bin/docker-compose, stdout:`

I would like to add that the value for the PATH environment variable is not the one reflected in the error message.
